test/test_multi.py

```diff
@@ -81,6 +82,13 @@ def test_multi_solve(self):
                 x_sparse = multi_solve(a_sparse, b_sparse, transpose=True)
                 x_dense = torch.linalg.solve(torch.eye(*a_dense.shape)-a_dense.T, b_dense)
                 self.assertTrue(self.compare_vectors(x_sparse, x_dense), (x_sparse, x_dense))
+    
+    ### WIP ###
```

Comment:
Does this still need work?

Reply:
This test is pretty basic and I think there should probably be some more testing. There isn't currently a great fgg in the test directory to test this function with. Do you think I should make a new fgg to test this with?

Reply:
Is example12p.json not good? I think it would be a lot of work for you to make a whole new FGG, but you could. Or we can build one from a PERPL program.

Reply:
example12.json and cycle.json both have a nonlinking nonterminal, but they are already at the end of a.shapes[0].keys() by default, so _order_nonterminals doesn't actually change anything when run on them.
I just added advanced_cycle.json which doesn't have this issue.